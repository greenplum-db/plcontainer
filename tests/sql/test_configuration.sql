CREATE ROLE pluser;

SET ROLE pluser;

CREATE OR REPLACE FUNCTION pyconf() RETURNS int4 AS $$
# container: plc_python_shared
return 10
$$ LANGUAGE plcontainer;

SET ROLE gpadmin;

SELECT pyconf();

SET ROLE gpadmin;

DROP FUNCTION pyconf();
DROP ROLE pluser;

-- Test non-exsited images
CREATE OR REPLACE FUNCTION py_no_exsited() RETURNS int4 AS $$
# container: plc_python_shared1
return 10
$$ LANGUAGE plcontainer;

SELECT py_no_exsited();
