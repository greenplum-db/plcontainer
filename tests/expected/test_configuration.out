select count(*) from plcontainer_containers_summary();
 count 
-------
     0
(1 row)

CREATE ROLE pluser;
NOTICE:  resource queue required -- using default resource queue "pg_default"
SET ROLE pluser;
CREATE OR REPLACE FUNCTION pyconf() RETURNS int4 AS $$
# container: plc_python_shared
return 10
$$ LANGUAGE plcontainer;
SET ROLE gpadmin;
SELECT pyconf();
 pyconf 
--------
     10
(1 row)

select count(*) from plcontainer_containers_summary();
 count 
-------
     1
(1 row)

SET ROLE pluser;
select count(*) from plcontainer_containers_summary();
 count 
-------
     0
(1 row)

SET ROLE gpadmin;
DROP FUNCTION pyconf();
DROP ROLE pluser;
